src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

```diff
@@ -1029,7 +1035,8 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 				} else {
 					this._instantiationService.invokeFunction(getDirectoryHistory)?.remove(e.item.label);
 				}
-			} else {
+				this.runRecent(type, filterMode, value);
+			} else if (e.button.iconClass === ThemeIcon.asClassName(Codicon.output)) {
```

Comment:
Could we handle this similar to `removeFromCommandHistoryButton` so we can do a reference check here instead of comparing the icon class?